### PR TITLE
Issue #SB-18222 feat: added support for shallow copy

### PR DESCRIPTION
**Pull request by krgauraw**

### CODE REVIEW

File: content-api/content-actors/src/main/scala/org/sunbird/content/util/CopyManager.scala [hunk 1]
```diff
@@ -71,21 +70,43 @@ object CopyManager {
...
+                    case CopyConstants.COPY_TYPE_SHALLOW => updateShallowHierarchy(request, node, originNode, originHierarchy)
+                    case _ => updateHierarchy(request,node, originNode, originHierarchy, copyType)
+                }
+            }).flatMap(f=>f)
         }).flatMap(f => f) recoverWith {case e: CompletionException => throw e.getCause}
     }
 
+    def updateHierarchy(request: Request, node: Node, originNode: Node, originHierarchy: util.Map[String, AnyRef], copyType:String)(implicit ec: ExecutionContext): Future[Node] = {
+        val updateHierarchyRequest = prepareHierarchyRequest(originHierarchy, originNode, node, copyType)
+        val hierarchyRequest = new Request(request)
```
Comment (PradyumnaNagendra) [inline content-api/content-actors/src/main/scala/org/sunbird/content/util/CopyManager.scala]: please add schema name as collection and  add version to request context

Comment (krgauraw) [reply content-api/content-actors/src/main/scala/org/sunbird/content/util/CopyManager.scala]: done

File: content-api/content-service/conf/routes [hunk 2]
```diff
@@ -30,7 +30,7 @@ DELETE    /content/v3/hierarchy/remove      controllers.v3.ContentController.rem
 PATCH     /content/v3/hierarchy/update      controllers.v3.ContentController.updateHierarchy
 GET       /content/v3/hierarchy/:identifier      controllers.v3.ContentController.getHierarchy(identifier:String, mode:Option[String])
 POST      /content/v3/upload/:identifier         controllers.v3.ContentController.upload(identifier:String)
-POST      /content/v3/copy/:identifier      controllers.v3.ContentController.copy(identifier:String, mode:Option[String])
+POST      /content/v3/copy/:identifier      controllers.v3.ContentController.copy(identifier:String, mode:Option[String], copyType:String ?= "deep")
```
Comment (maheshkumargangula) [inline content-api/content-service/conf/routes]: the path parameter should be `type` not `copyType`. 
Ex: /content/v3/copy/do_123?type=shallow

Comment (krgauraw) [reply content-api/content-service/conf/routes]: done

File: content-api/content-actors/src/main/scala/org/sunbird/content/util/CopyManager.scala (hunk 1 above)
Comment (krgauraw) [reply content-api/content-actors/src/main/scala/org/sunbird/content/util/CopyManager.scala]: done

**Pull request merged**